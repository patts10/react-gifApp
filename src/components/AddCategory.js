import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {

  const [inputvalue, setInputvalue] = useState("");

  const handleInputChange = (e) => {
    setInputvalue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputvalue.trim().length > 2) {
      setCategories((c) => [inputvalue, ...c]);
      setInputvalue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        className="my-5"
        type="text"
        value={inputvalue}
        onChange={handleInputChange}
        placeholder={"Search..."}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
