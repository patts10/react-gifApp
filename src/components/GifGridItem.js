import React from 'react';

export const GifGridItem = ( {title, url }) => {


  return (
    <div className='overflow-hidden rounded-md border-black border-2 xs:w-1/2 md:w-2/12 animate__animated animate__fadeIn mx-px my-5'>
      <img 
        className='w-full md:h-5/6 sm:h-5/6 xs:h-5/6'
        src={ url }
        alt={ title }
      />
      <p className='my-1 text-center'>{ title }</p>
    </div>
  );
};
