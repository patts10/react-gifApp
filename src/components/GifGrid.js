import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {

  const { data:images, loading } = useFetchGifs( category );

  

  return (
    <div className="bg-slate-100 p-5">
      <h3 className="text-lg font-bold text-gray-900 sm:text-3xl sm:truncate mt-3 mb-5 mt-5"> { category } </h3>

      { loading && <p className="animate__animated animate__flash">Loading...</p>}

      <div>
        <ol className="flex flex-wrap justify-around">
          {images.map((img) => (
            <GifGridItem key={img.id} {...img} />
          ))}
        </ol>
      </div>
    </div>
  );
};