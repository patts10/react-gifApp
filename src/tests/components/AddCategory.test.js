
import { shallow } from 'enzyme';
import { AddCategory } from "../../components/AddCategory";



describe('Tests on <AddCategory />', () => {

  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  
  beforeEach( () => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  })
  
  test('should show it correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  // test('should change input text', () => {
    
  //   const value = 'patts';
  //   const input = wrapper.find('input');

  //   input.simulate('change', { target: { value } });
  // });
  
  test('should not post information when submit', () => {
    
    wrapper.find('form').simulate('submit',{ preventDefault(){} });
    
    expect( setCategories ).not.toHaveBeenCalled();

  });

  test('should call setCategories and clean input text', () => {
    
    const value = 'hello world'
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect( setCategories ).toHaveBeenCalled();
    expect( setCategories ).toHaveBeenCalledTimes(1);
    expect( setCategories ).toHaveBeenCalledWith( expect.any(Function));

    const inputValue = wrapper.find('input');
    expect(inputValue.prop('value')).toBe('');
  });
  
  
});
