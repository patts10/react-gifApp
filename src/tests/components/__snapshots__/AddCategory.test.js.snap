// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AddCategory /> should show it correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="my-5"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</form>
`;
